### DISCUSSION TITLE
[Question] Possible to test out pre-release typescript versions?

### DISCUSSION THREAD

**Discussion by JasonKleban** [yarnpkg/berry#2798] [Q&A] [answered]
It is my understanding (possibly dead wrong) that yarn patches typescript with a tsc-version-specific patch to override typescript's private module resolution strategy.  Is that true?  Where is the code for these patches??

Most importantly, if there is a pre-release version of typescript that we want to test in a yarn 2 monorepo, are we out of luck?  I get "Cannot apply hunk ...." during `yarn install` if I try to `resolve` to a new version which I figure is because the patch can't be applied to an unrecognized tsc.

I'd love an answer, and but this is secondarily a request for the above topics to be documented.  Thanks!

**Comment by arcanis** [comment, marked as answer]
> It is my understanding (possibly dead wrong) that yarn patches typescript with a tsc-version-specific patch to override typescript's private module resolution strategy. Is that true? Where is the code for these patches??

In the TypeScript repo: https://github.com/microsoft/TypeScript/pull/35206

> Most importantly, if there is a pre-release version of typescript that we want to test in a yarn 2 monorepo, are we out of luck? I get "Cannot apply hunk ...." during yarn install if I try to resolve to a new version which I figure is because the patch can't be applied to an unrecognized tsc.

This will be solved starting from our next release, the 3.0 (the fix is already in master and can be tried out with `yarn set version 3.0.0-rc.2`). Should conflicts happen, you'll just have to switch your install strategy to `node-modules` until we can update the patch.
